// yandex-translate-async/dist/lib/YandexTranslate.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _axios = require('axios');

var _axios2 = _interopRequireDefault(_axios);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var YandexTranslate = function () {
  function YandexTranslate(_ref) {
    var apiKey = _ref.apiKey;

    _classCallCheck(this, YandexTranslate);

    this.apiKey = apiKey;
  }

  _createClass(YandexTranslate, [{
    key: 'translate',
    value: async function translate() {
      var text = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
      var opts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : { langIn: 'en', langOut: 'zh' };

      try {
        var langIn = opts.langIn,
            langOut = opts.langOut;


        var res = await (0, _axios2.default)({
          method: 'post',
          url: 'https://translate.yandex.net/api/v1.5/tr.json/translate',
          params: {
            key: this.apiKey,
            lang: langIn + '-' + langOut,
            text: text,
            format: 'plain'
          }
        });

        var outputText = res.data.text;

        return outputText[0];
      } catch (e) {
        console.error('An error occured while translating.');
        console.error(e);
      }
      return null;
    }
  }]);

  return YandexTranslate;
}();

exports.default = YandexTranslate;
